projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/asm/LLVMAMD64CpuidNode.java

```diff
@@ -29,20 +29,60 @@
  */
 package com.oracle.truffle.llvm.nodes.asm;
 
+import com.oracle.truffle.api.CompilerAsserts;
 import com.oracle.truffle.api.dsl.NodeChild;
 import com.oracle.truffle.api.dsl.Specialization;
 import com.oracle.truffle.api.frame.VirtualFrame;
-import com.oracle.truffle.llvm.nodes.asm.support.LLVMAMD64WriteRegisterNode.LLVMAMD64WriteI32RegisterNode;
+import com.oracle.truffle.llvm.nodes.asm.support.LLVMAMD64WriteValueNode;
 import com.oracle.truffle.llvm.runtime.nodes.api.LLVMExpressionNode;
 
 @NodeChild("level")
 public abstract class LLVMAMD64CpuidNode extends LLVMExpressionNode {
-    @Child private LLVMAMD64WriteI32RegisterNode eax;
-    @Child private LLVMAMD64WriteI32RegisterNode ebx;
-    @Child private LLVMAMD64WriteI32RegisterNode ecx;
-    @Child private LLVMAMD64WriteI32RegisterNode edx;
+    public static final String BRAND = "Sulong"; // at most 48 characters
+    public static final String VENDOR_ID = "SulongLLVM64"; // exactly 12 characters
 
-    public LLVMAMD64CpuidNode(LLVMAMD64WriteI32RegisterNode eax, LLVMAMD64WriteI32RegisterNode ebx, LLVMAMD64WriteI32RegisterNode ecx, LLVMAMD64WriteI32RegisterNode edx) {
+    public static final int[] BRAND_I32 = getI32(BRAND, 12);
+    public static final int[] VENDOR_ID_I32 = getI32(VENDOR_ID, 3);
+
+    private static int[] getI32(String s, int len) {
+        CompilerAsserts.neverPartOfCompilation();
+        int[] i32 = new int[len];
+        for (int i = 0; i < len; i++) {
+            byte b1 = getI8(s, i * 4);
+            byte b2 = getI8(s, i * 4 + 1);
+            byte b3 = getI8(s, i * 4 + 2);
+            byte b4 = getI8(s, i * 4 + 3);
+            i32[i] = Byte.toUnsignedInt(b1) | Byte.toUnsignedInt(b2) << 8 | Byte.toUnsignedInt(b3) << 16 | Byte.toUnsignedInt(b4) << 24;
+        }
+        return i32;
+    }
+
+    private static byte getI8(String s, int offset) {
+        CompilerAsserts.neverPartOfCompilation();
+        if (offset >= s.length()) {
+            return 0;
+        } else {
+            return (byte) s.charAt(offset);
+        }
+    }
+
+    @Child private LLVMAMD64WriteValueNode eax;
+    @Child private LLVMAMD64WriteValueNode ebx;
+    @Child private LLVMAMD64WriteValueNode ecx;
+    @Child private LLVMAMD64WriteValueNode edx;
+
+    // FN=1: EDX
+    public static final int TSC = 1 << 4;
+    // FN=1: ECX
+    public static final int RDRND = 1 << 30;
+    // FN=7/0: EBX
+    public static final int RDSEED = 1 << 18;
```

Comment:
I think names like `RDSEED_IS_SUPPORTED` makes it easier to understand what these constants are used for.